### PR TITLE
*: remove useless cache in the MergePartitionStats2GlobalStats

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/47219

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/globalstats/global_stats.go [hunk 1]
```diff
@@ -122,27 +121,21 @@ func MergePartitionStats2GlobalStats(
...
+				globalStats.MissingPartitionStats = append(globalStats.MissingPartitionStats, fmt.Sprintf("partition `%s`", def.Name.L))
+				continue
 			}
-			allPartitionStats[partitionID] = partitionStats
+			err = err1
+			return
 		}
 
 		for i := 0; i < globalStats.Num; i++ {
+			// GetStatsInfo will return the copy of the statsInfo, so we don't need to worry about the memory problem.
```
Comment (time-and-fate) [inline statistics/handle/globalstats/global_stats.go:137]: What "memory problem"? Please be a bit more specific.

**Review by time-and-fate — APPROVED** [review]

File: statistics/handle/globalstats/global_stats.go (hunk 1 above)
Comment (hawkingrei) [reply statistics/handle/globalstats/global_stats.go]: I have updated. it has been changed as ```data race```.

**Review by winoros — APPROVED** [review]

**Pull request merged**